core/queryparser/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/TupleExprBuilder.java

```diff
@@ -601,15 +601,6 @@ public TupleExpr visit(ASTSelect node, Object data) throws VisitorException {
 				Var projVar = (Var) child.jjtAccept(this, null);
 				ProjectionElem elem = new ProjectionElem(projVar.getName());
 				projElemList.addElement(elem);
-
-				VarCollector whereClauseVarCollector = new VarCollector();
-				result.visit(whereClauseVarCollector);
-
-				if (!whereClauseVarCollector.collectedVars.contains(projVar)) {
-					ExtensionElem extElem = new ExtensionElem(projVar, projVar.getName());
-					extension.addElement(extElem);
-					elem.setSourceExpression(extElem);
-				}
 			} else {
```

Comment:
Removing the lines above here is the core of this fix. Everything else in the diff (apart from the added test) is just cleanup.